File: frontend/src/types/Product.ts
export interface Product {
  id: number;
  name: string;
  description: string;
  sku: string;
  price: number;
  quantityInStock: number;
  createdAt: string;
  updatedAt: string;
}

export interface CreateProductDto {
  name: string;
  description: string;
  sku: string;
  price: number;
  quantityInStock: number;
}

export interface UpdateProductDto {
  name: string;
  description: string;
  price: number;
  quantityInStock: number;
}

File: src/DemoInventory.Domain/Entities/Product.cs
namespace DemoInventory.Domain.Entities;

public class Product
{
    public int Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Description { get; set; } = string.Empty;
    public string SKU { get; set; } = string.Empty;
    public decimal Price { get; set; }
    public int QuantityInStock { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
}